### PR TITLE
Add preview build change functionality to EquipBuildModal

### LINKED ISSUE #1619: Change the "Equip Build" confirmation pop-up to provide a preview of changes to the currently active build
While the currently active build is displayed on the character card at the top of the page and a user can technically compare a generated build with the active build by scrolling back and forth before equipping, it would be more convenient to modify the current confirmation popup when the user clicks "Equip Build" to additionally display this information as a preview. Probably something along these lines with the current confirmation text and buttons at the bottom:
image

### PR DESCRIPTION
**Pull request by arquelion**
## Describe your changes
- Moved `EquipBuildModal` from `BuildReal.tsx` into a separate file - Both the equip button from build management and the equip button on the optimize tab do functionally the same thing, so this allows both to utilize `EquipBuildModal` for this purpose
    - Modified `EquipBuildModal` to display a preview of build changes above the confirmation and copy equipped checkbox
    - Added a new prop `equipChangeProps` containing the necessary name and weapon/artifact information to be displayed in the preview
    - Changed several strings to automatically insert the correct build name instead of always using "Equipped"
- Changed the relevant call sites in `BuildDisplayItem` and `BuildReal` to use the modified version of `EquipBuildModal`

## Issue or discord link

- Resolves #1619 

## Testing/validation
The equipment swap preview when only a weapon is equipped
image

The equipment swap preview on a narrower window
image

Equip preview and confirmation from the optimize tab

https://github.com/frzyc/genshin-optimizer/assets/47800040/97569633-097e-40ac-b41a-4e3eb874aba5

Equip preview and confirmation from the loadout settings modal

https://github.com/frzyc/genshin-optimizer/assets/47800040/0d0e2f1c-3cd0-4e6c-a0f9-9c14bff88166

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by arquelion** [comment]
`EquipBuildModal` seems to be a little too far to the right (or Loadout Settings is a little too far to the left). It's also like this on live, but the modal being wider makes this significantly more obvious. Probably a value missing or needing to be added somewhere...

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            display: 'flex',
+            flexDirection: 'column',
+          }}
+        >
+          <Box>
+            {/* Active Build */}
+            <CardThemed bgt="light">
+              <CardHeader
+                sx={{ pb: 0 }}
+                title={`${equipChangeProps.currentName}: Current Equipment`}
```
Comment (frzyc) [inline apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx:96]: I'd suggest using a badge to show information:
`<Sqbadgecolor="warning">Current Equipment</Sqbadge>`

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+              display="flex"
+              justifyContent="center"
+              alignItems="center"
+              padding={2}
+            >
+              <KeyboardArrowDown sx={{ fontSize: 40 }} />
+            </Box>
+            {/* New Build */}
+            <CardThemed bgt="light">
+              <CardHeader sx={{ pb: 0 }} title={'New Equipment'} />
```
Comment (frzyc) [inline apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx:138]: Same as above, add 
`<Sqbadge color="success">new Equipment<Sqbadge>`

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+              </CardContent>
+            </CardThemed>
+          </Box>
+        </CardContent>
+        <CardContent sx={{ pt: '0', display: 'flex', flexDirection: 'column' }}>
+          <Typography sx={{ fontSize: 20 }}>
+            Do you want to make the changes shown above?
+          </Typography>
+          <FormControlLabel
+            label={`Copy the current equipment in ${equipChangeProps.currentName} to a new build. Otherwise, they will be overwritten.`}
```
Comment (frzyc) [inline apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx:175]: Bold the name
`<strong>{equipChangeProps.currentName}</strong>`

**Review by frzyc — CHANGES_REQUESTED** [review]
Some minor changes to emphasize equipment change.

👍 On merging the equipment on the two UIs (build and gen build).

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx (hunk 1 above)
Comment (arquelion) [reply apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx]: done

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx (hunk 3 above)
Comment (arquelion) [reply apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx]: done

File: apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx (hunk 2 above)
Comment (arquelion) [reply apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx]: done

**Comment by arquelion** [comment]
Changes made, I also moved the build name out of the preview and up into the header and bolded it there as well:
image

I wasn't a terribly big fan of having the build name tacked onto the "Current Equipment" line but unsure if just having it between the header and the card would be better than both of these.

**Review by frzyc — APPROVED** [review]

**Pull request merged**